src/NDS.cpp

```diff
@@ -907,7 +907,12 @@ bool DoSavestate(Savestate* file)
     }
 
     for (int i = 0; i < 8; i++)
-        DMAs[i]->DoSavestate(file);
+    {
+        if (!DMAs[i]->DoSavestate(file))
```

Comment:
the DMA objects can't really be null except for OOM.

They could probably be packed directly as values instead of pointers.

Reply:
How's this?

Reply:
how's what?

Reply:
The [two](https://github.com/melonDS-emu/melonDS/pull/1856/commits/66d9573608fa4ab2f3e5070688c622716e52133d) [commits](https://github.com/melonDS-emu/melonDS/pull/1856/commits/3ef38f4247f56e4d6f172b1bd87efe77ee539a50) that follow this one where I address this feedback.